spring-restdocs/src/main/java/org/springframework/restdocs/hypermedia/LinkExtractors.java

```diff
@@ -70,10 +65,13 @@ public static LinkExtractor atomLinks() {
 	 * @return The extractor for the content type, or {@code null}
 	 */
 	public static LinkExtractor extractorForContentType(String contentType) {
-		if (MediaType.APPLICATION_JSON_VALUE.equals(contentType)) {
+		if (null == contentType) {
+			return null;
+		}
+		if (contentType.startsWith(MediaType.APPLICATION_JSON_VALUE)) {
 			return atomLinks();
 		}
-		else if ("application/hal+json".equals(contentType)) {
+		else if (contentType.startsWith("application/hal+json")) {
```

Comment:
As mentioned in the main comments on the PR, I think this should use `MediaType.parseMediaType(contentType)`